src/cryocanvas/app.py

```diff
@@ -128,19 +131,22 @@ def _add_widget(self):
         self._connect_events()
 
     def _connect_events(self):
-        # Use a partial function to pass additional arguments to the event handler
-        on_data_change_handler = tz.curry(self.on_data_change)(app=self)
         for listener in [
             self.viewer.camera.events,
             self.viewer.dims.events,
-            self.painting_layer.events.paint,
         ]:
             listener.connect(
                 debounced(
-                    ensure_main_thread(on_data_change_handler),
+                    self.on_view_change,
                     timeout=1000,
                 )
             )
+        self.painting_layer.events.paint.connect(
```

Comment:
I separate out the paint event response (`on_data_change`) from the camera/dims response (`on_view_change`) since with default parameter values (i.e. with `Whole Image`) we don't need to update anything when the view changes.

This might be worth bringing out into a smaller PR if these defaults are expected to be used in the long term.